new tutorial for find the easy pass of hackthebox challenges. (#63)
rizinorg/book: src/crackmes/hackthebox/find-the-easy-pass/fire-up-the-debugger.md

# Fire up the debugger

> The next step will be to put a breakpoint at 0x0045412b, press `F2` key

Reviewer: Same issue here, why at that address?

Author: I explained why this breakpoint with the last changes.

# Fire up the debugger

The next step will be to put a breakpoint at 0x0045412b,

> this address contains probably the right password or our.

Reviewer 2: I think you should get all of these addresses/offsets inside single backticks.

Author: Sure, I will do.

> this address contains probably the right password or our.

Reviewer 2: Looks like this sentence got over in the middle?

Author: Yes, but the rendering seems correct  :
![rizin-page-served-01](https://user-images.githubusercontent.com/85988120/124504105-1118cb00-ddc7-11eb-825a-3f21922da6b9.png)

Reviewer 2: Sorry, I should've been more clear on what I was pointing to.

What I was referring to is the line `this address contains probably the right password or our.`
Or our what? -- I think that sentence is sort of incomplete.

![run-the-program-debugged-4](./img/fire-up-the-debugger-04.png)

The program is launched.

![run-the-program-debugged-5](./img/fire-up-the-debugger-05.png)

Enter a placeholder text and click on the "Check Password" button.

Logically, the program should stop at our breakpoint point.

![run-the-program-debugged-6](./img/fire-up-the-debugger-06.png)

Press `q` and then `p` to show the debugger view panel.

![run-the-program-debugged-7](./img/fire-up-the-debugger-07.png)

Press `F8` key to step over.

![run-the-program-debugged-8](./img/fire-up-the-debugger-08.png)

Press `g`, enter `eax` and `P` twice.

![run-the-program-debugged-9](./img/fire-up-the-debugger-09.png)

That's great our placeholder text is in memory.

Press `p` twice, `g` and enter `rip`.

![run-the-program-debugged-10](./img/fire-up-the-debugger-10.png)

Press `F8` key.

![run-the-program-debugged-11](./img/fire-up-the-debugger-11.png)

Press `g`, enter `edx` and `P` twice.

![run-the-program-debugged-12](./img/fire-up-the-debugger-12.png)

The right password seems to be **fortan!**.

Press `p` twice, `g` and enter `rip`.

Press `F8` key twice.

![run-the-program-debugged-13](./img/fire-up-the-debugger-13.png)

> The call at `0x0045413d` seems compare the previous password and if not
> correct the program jump on the "Wrong Pasword!" message.

Reviewer 2: ```suggestion
The call at `0x0045413d` seems to compare the previous password and 
when unequal, the program jumps to the "Wrong Pasword!" message.
```

Press `F9` key to show the message box.

After that, enter "fortan!" string and repeat the previous step until
the address `0x0045413d`.

![run-the-program-debugged-14](./img/fire-up-the-debugger-14.png)

Press `F8`key twice.

![run-the-program-debugged-15](./img/fire-up-the-debugger-15.png)

> 
> it's seems all right ! Press `F9` key to confirm this fact.

Reviewer 2: ```suggestion

It seems all right! Press `F9` key to confirm this fact.
```
